### PR TITLE
Fix voting authority and locked by tickets balance calculation

### LINKED ISSUE #821: getbalance showing errorneously between lockedbytickets, votingauthority and total.
*Using dcrwallet 1.0.3/1.0.5*

I think this issue has been created somewhere but i thought of submitting some data output in assisting the bug squashing.

I'm doing PoS via "VPS hot voting wallet" and also "Pool mining". These are my getbalance output :

On VPS:  
{
     "accountname": "default",
     "immaturecoinbaserewards": 0,
     "immaturestakegeneration": 0,
     "lockedbytickets": 0,
     "spendable": 0,
     "total": 8590.25381921,
     "votingauthority": 8620.83221921
   },

On my main wallet which i imported script from the pool :
 {
     "accountname": "imported",
     "immaturecoinbaserewards": 0,
     "immaturestakegeneration": 0,
     "lockedbytickets": 2175.57874403,
     "spendable": 0,
     "total": 2174.24083072,
     "votingauthority": 2179.08828322
   }

**My observation:**

1. On both cases, my voting authority is not the same as my total. 
2. In addtion, my "lockbytickets" on the pool also differs from "total"
3. Running --rescanwallet does not fix the problem
4. I remember that these number used to tally with each other in the past. It only start appearing from 1.0.3 onwards.
5. Running dcrctl --wallet listunspent and adding up all the numbers manually confirmed that only the "Total" shows the correct figure.

### PR DESCRIPTION
**Pull request by alexlyp**
Since they use the sstx commitments to calculate fees need to be taken into account.
Fixes #821

### CODE REVIEW

File: wallet/udb/txmined.go [hunk 1]
```diff
@@ -3594,7 +3599,7 @@ func (s *Store) balanceFullScan(ns, addrmgrNs walletdb.ReadBucket, minConf int32
 				ab.VotingAuthority += utxoAmt
 			}
 			ab.LockedByTickets += utxoAmt
-
+			fmt.Println("here", utxoAmt)
```
Comment (jrick) [inline wallet/udb/txmined.go]: Needs removal

File: wallet/udb/txmined.go [hunk 2]
```diff
@@ -3594,7 +3599,6 @@ func (s *Store) balanceFullScan(ns, addrmgrNs walletdb.ReadBucket, minConf int32
 				ab.VotingAuthority += utxoAmt
 			}
 			ab.LockedByTickets += utxoAmt
-
```
Comment (dajohi) [inline wallet/udb/txmined.go]: Does this minConf==0 part need the same change as above?

Comment (alexlyp) [inline wallet/udb/txmined.go]: hrm probably

**Comment by raedah** [comment]
I have tested this patch and verified that it resolves the behavior described in https://github.com/decred/dcrwallet/issues/850 . After a rescanwallet, all balances match up on both the ticketbuyer machine as well as the voting box, and also match up with a wallet freshly built from seed.

File: wallet/udb/txdb.go [hunk 3]
```diff
@@ -790,6 +790,14 @@ func extractRawCreditIndex(k []byte) uint32 {
 	return byteOrder.Uint32(k[68:72])
 }
 
+func extractRawUnminedTx(k []byte) []byte {
```
Comment (jrick) [inline wallet/udb/txdb.go]: this should be v not k, it is passed the value not the key.

**Pull request merged**